setup.py

```diff
@@ -121,14 +119,37 @@ def run(self):
                 # e.g. -framework GDAL
                 extra_link_args.append(item)
         gdalversion = gdal_output[3]
+        if gdalversion:
+            log.info("GDAL API version obtained from gdal-config: %s",
+                     gdalversion)
 
     except Exception as e:
         if os.name == "nt":
             log.info("Building on Windows requires extra options to setup.py "
                      "to locate needed GDAL files.\nMore information is "
                      "available in the README.")
         else:
-            log.warning("Failed to get options via gdal-config: %s", str(e))
+            log.warn("Failed to get options via gdal-config: %s", str(e))
+
+    # Get GDAL API version from environment variable.
+    gdalversion = os.environ.get('GDAL_VERSION', gdalversion)
+    if gdalversion:
+        log.info("GDAL API version obtained from command line option: %s",
```

Comment:
@sgillies Typo in the log message - should state the version was obtained from the env variable.